edc-extensions/agreements/retirement-evaluation-core/src/testFixtures/java/org/eclipse/tractusx/edc/agreements/retirement/store/AgreementsRetirementStoreTestBase.java

```diff
@@ -0,0 +1,106 @@
+ * Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.agreements.retirement.store;
+
+import org.eclipse.edc.spi.query.Criterion;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.store.AgreementsRetirementStore;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.types.AgreementsRetirementEntry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class AgreementsRetirementStoreTestBase {
+
+    private AgreementsRetirementStore store;
+
+    @BeforeEach
+    void setup() {
+        store = getStore();
+    }
+
+    @Test
+    void findRetiredAgreement() {
+        var agreementId = "test-agreement-id";
+        var entry = createRetiredAgreementEntry(agreementId, "mock-reason");
+        store.save(entry);
+
+        var query = createFilterQueryByAgreementId(agreementId);
+        var result = store.findRetiredAgreements(query);
+        assertThat(result.getContent())
+                .isNotNull()
+                .hasSize(1)
+                .first()
+                .extracting(AgreementsRetirementEntry::getAgreementId)
+                .isEqualTo(agreementId);
+    }
+
+    @Test
+    void findRetiredAgreement_notExists() {
+        var agreementId = "test-agreement-not-exists";
+        var query = createFilterQueryByAgreementId(agreementId);
+        var result = store.findRetiredAgreements(query);
+        assertThat(result.getContent()).isEmpty();
+    }
+
+    @Test
+    void save_whenExists() {
+        var entry = createRetiredAgreementEntry("test-agreement-id", "mock-reason");
+        store.save(entry);
+        assertThat(store.save(entry).succeeded()).isFalse();
+    }
+
+    @Test
+    void delete() {
+        var agreementId = "test-agreement-id";
+        var entry = createRetiredAgreementEntry(agreementId, "mock-reason");
+        store.save(entry);
+        var delete = store.delete(agreementId);
+        assertThat(delete.succeeded()).withFailMessage(delete::getFailureDetail).isTrue();
```

Comment:
the ```getFailureDetail```should be empty, right?

Reply:
I refactored the test because the `withFailMessage()` does not have the effect I expected it to have. Thanks for pointing this out.
[70c1003](https://github.com/eclipse-tractusx/tractusx-edc/pull/1615/commits/70c10038ac71342bf0d97d525b0962a0f0a2be6f)